src/MassTransit/Context/BaseReceiveEndpointOutboxTransportContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿using MassTransit.Configuration;
+using MassTransit.Transports.Outbox;
+
+namespace MassTransit.Context
+{
+    public abstract class BaseReceiveEndpointOutboxTransportContext : BaseReceiveEndpointContext
+    {
+        private readonly IHostConfiguration _hostConfiguration;
+
+        public BaseReceiveEndpointOutboxTransportContext(IHostConfiguration hostConfiguration, IReceiveEndpointConfiguration configuration)
+            : base(hostConfiguration, configuration)
+        {
+            _hostConfiguration = hostConfiguration;
+        }
+
+        protected override IPublishTransportProvider CreateDecoratedPublishTransportProvider()
+        {
+            var transport = CreatePublishTransportProvider();
+
+            if (_hostConfiguration.UseOutbox)
```

Comment:
@phatboyg not sure if you'll like this approach to decorate the transport here. I'm open to alternatives.

Reply:
I'll have to look through it all tomorrow night at the earliest, excited to check it out though.